### PR TITLE
Make pod/container-kill controlled by twophase controller in scheduler mode

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 

This requests make container/pod-kill chaos only to maintain their own logic without where the calling from in order to support trigger chaos 
And I also let the pod/container-kill chaos controlled by twophase controller by hacking the recover duration as 0 second when duration is nil.

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

**Comment by zhouqiang-cl** [comment]
> And I also think the logic in specific chaos action controller should only contain performing the chaos action without caring where the calling from.

I think so. Now we add a new chaos, then need to add two types of chaos, we should make it more simple

**Comment by cwen0** [comment]
@Yisaer If you remove these checks, the scheduler of chaos object won't work.

**Comment by cwen0** [comment]
> @Yisaer If you remove these checks, the scheduler of chaos object won't work.

It is a bug here, Can you fix it?

**Comment by Yisaer** [comment]
> @Yisaer If you remove these checks, the scheduler of chaos object won't work.

@cwen0 Thanks for reminding me. I previously had a misunderstanding for `container-kill` and `pod-kill` reconciler.
It seems we can't use twophase controller to directly control the `pod/container`-killl action yet. However, I think there still exists the way to achieve it. I would try to figure it out.

**Comment by Yisaer** [comment]
ref https://github.com/pingcap/chaos-mesh/issues/246

**Comment by Yisaer** [comment]
@zhouqiang-cl @cwen0  PTAL

File: controllers/twophase/types.go [hunk 1]
```diff
@@ -83,11 +83,16 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	}
 
 	scheduler := chaos.GetScheduler()
-	if duration == nil || scheduler == nil {
-		r.Log.Info("scheduler and duration should be defined currently")
+	if scheduler == nil {
+		r.Log.Info("scheduler should be defined currently")
```
Comment (cwen0) [inline controllers/twophase/types.go]: For `container-kill` and `pod-kill`, this is a normal log, for the other action, this is an error.

Comment (Yisaer) [reply controllers/twophase/types.go]: Not actually, the twophase controller let the `duration` become 0 second when it is omitted which would take effect nearly 10 seconds as default resync interval.
And further more, we can validate the chaos Spec in their own chaos kind controller or validation webhook when created or updated.

Comment (cwen0) [reply controllers/twophase/types.go]: Sorry, I misunderstand this check. I think we need to return `ctrl.Result{RequeueAfter: recover-duration}` when to set recover time.

**Comment by cwen0** [comment]
@YangKeao PTAL

**Pull request merged**